### ISSUE TITLE
v4l2loopback

### ISSUE THREAD

**Issue by jmlipari** [linux-surface/linux-surface#218]
Hello,
I'm newbie with Linux systems, and I really enjoying the speed and stability of the OS.
The only thing that it's really annoying for me is the webcam and videoconferencing software, so I planned to install "Iriun Webcam" in order to use my cellphone camera as webcam and workaround the current limitations.
I did it, but software needs v4l2loopback. I downloaded, compiled, and loaded to kernel but it seems to be not working.
My video devices **BEFORE** modprobe
jmlipari@jmlipari-Surface-Book:~/v4l2loopback-master$ ls -ltrh /dev/video*
crw-rw----+ 1 root video 81,  1 Jun 23 07:22 /dev/video1
crw-rw----+ 1 root video 81,  2 Jun 23 07:22 /dev/video2
crw-rw----+ 1 root video 81,  0 Jun 23 07:22 /dev/video0
crw-rw----+ 1 root video 81,  3 Jun 23 07:22 /dev/video3
crw-rw----+ 1 root video 81,  8 Jun 23 07:22 /dev/video4
crw-rw----+ 1 root video 81, 12 Jun 23 07:22 /dev/video8
crw-rw----+ 1 root video 81, 10 Jun 23 07:22 /dev/video6
crw-rw----+ 1 root video 81, 13 Jun 23 07:22 /dev/video9
crw-rw----+ 1 root video 81, 14 Jun 23 07:22 /dev/video10
crw-rw----+ 1 root video 81, 11 Jun 23 07:22 /dev/video7
crw-rw----+ 1 root video 81,  9 Jun 23 07:22 /dev/video5
crw-rw----+ 1 root video 81, 15 Jun 23 07:22 /dev/video11
crw-rw----+ 1 root video 81, 17 Jun 23 07:22 /dev/video13
crw-rw----+ 1 root video 81, 16 Jun 23 07:22 /dev/video12

jmlipari@jmlipari-Surface-Book:~/v4l2loopback-master$ **sudo modprobe v4l2loopback devices=2**
jmlipari@jmlipari-Surface-Book:~/v4l2loopback-master$ ls -ltrh /dev/video*
crw-rw----+ 1 root video 81,  1 Jun 23 07:22 /dev/video1
crw-rw----+ 1 root video 81,  2 Jun 23 07:22 /dev/video2
crw-rw----+ 1 root video 81,  0 Jun 23 07:22 /dev/video0
crw-rw----+ 1 root video 81,  3 Jun 23 07:22 /dev/video3
crw-rw----+ 1 root video 81,  8 Jun 23 07:22 /dev/video4
crw-rw----+ 1 root video 81, 12 Jun 23 07:22 /dev/video8
crw-rw----+ 1 root video 81, 10 Jun 23 07:22 /dev/video6
crw-rw----+ 1 root video 81, 13 Jun 23 07:22 /dev/video9
crw-rw----+ 1 root video 81, 14 Jun 23 07:22 /dev/video10
crw-rw----+ 1 root video 81, 11 Jun 23 07:22 /dev/video7
crw-rw----+ 1 root video 81,  9 Jun 23 07:22 /dev/video5
crw-rw----+ 1 root video 81, 15 Jun 23 07:22 /dev/video11
crw-rw----+ 1 root video 81, 17 Jun 23 07:22 /dev/video13
crw-rw----+ 1 root video 81, 16 Jun 23 07:22 /dev/video12
**crw-rw----+ 1 root video 81, 21 Jun 23 09:59 /dev/video15
crw-rw----+ 1 root video 81, 20 Jun 23 09:59 /dev/video14**

As you see above, the video devices has been created, but any application "see" the devices.
Iriunweb app says: "initialization failed. You may want to run sudo modprobe v4l2loopback exclusive_caps=1"

I think this is a kernel related problem but my linux skills are very low but the approach to workaround the problem with surface cam is valid. Any help?

My Environment
- Hardware model: Surface Book 1
- Kernel version: 5.7.4-surface
- Distribution: Elementary OS

**Comment by jhand2** [comment]
See https://github.com/linux-surface/linux-surface/issues/91 for more information on camera support.

tl;dr, webcams are not yet supported for many of the surface devices.

**Comment by jmlipari** [comment]
> See #91 for more information on camera support.
> 
> tl;dr, webcams are not yet supported for many of the surface devices.

Are video Virtual devices supported? I'm trying to use my phone as camera.
Thank you

**Comment by jhand2** [comment]
@jmlipari my apologies, I missed that portion of your comment. I'm not familiar with this particular issue.

**Comment by danieldeng2** [comment]
droidcam works well enough for me?

**Comment by jmlipari** [comment]
> droidcam works well enough for me?

I'll give it a try. Thank you

**Comment by jmlipari** [comment]
Thank you @danieldeng2 . Finally solved my problem using Droidcam software. I've found better optimized for Linux systems than Iriunwebcam. Now, **I can make Google Meet videocalls using my phone as webcam in a Surface book device**. The recipe:

    Elementary OS 5.1 Hera
    Surface-linux Kernel (https://github.com/linux-surface/linux-surface)
    v4l2loopback 0.12.5 (https://github.com/umlaeute/v4l2loopback)
    Droidcam (https://www.dev47apps.com/)

I hope this helps. Thank you!
